new_sem10_oop/dz1.py

```diff
@@ -0,0 +1,31 @@
+# Доработаем задачи 5-6. Создайте класс-фабрику. 
+# Класс принимает тип животного (название одного из созданных классов) и параметры для этого типа. 
+# Внутри класса создайте экземпляр на основе переданного типа и верните его из класса-фабрики.
+
+import inspect
+from pack10.Animal import Fish, Bird, Mammal
+
+class Fabric:
+
+    @classmethod
+    def create_animal_from_type(cls, animal_type: type):
+        res = inspect.getfullargspec(animal_type.__init__)
+        res = dict(map(lambda arg: (arg, input(f'please input {arg}: ')), res.args[1:]))
+        res = animal_type(**res)
+        return res
```

Comment:
Зачем же такие сложности?
```suggestion
    def create_animal_from_type(cls, animal_type: type, *args):
        return animal_type(*args)
```